phabricator.js

```diff
@@ -0,0 +1,129 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+let parentRevisionPromise;
+
+async function injectButton(block) {
+  let path = block.querySelector('h1.differential-file-icon-header').textContent;
+  let buttonDiv = block.querySelector('div.differential-changeset-buttons');
+
+  function getLines() {
+    return block.querySelectorAll('table.differential-diff tbody tr th:first-child');
+  }
+
+  let button = document.createElement('button');
+  button.type = 'button';
+  button.textContent = 'Loading Code Coverage...';
+  button.disabled = true;
+  button.className = 'button button-grey';
+  buttonDiv.append(button);
+
+  let parentRevision = await parentRevisionPromise;
+  let data;
+  if (!parentRevision) {
+    data = null;
+  } else {
+    ({ data } = await fetchCoverage(parentRevision, path));
```

Comment:
Before retrieving the coverage for a file, we should check `isCoverageSupported`